w3af/plugins/grep/analyze_cookies.py

```diff
@@ -204,13 +204,16 @@ def _http_only(self, request, response, cookie_obj,
         :return: None
         """
         if not self.HTTPONLY_RE.search(cookie_header_value):
-            
             vuln_severity = severity.MEDIUM if fingerprinted else severity.LOW
-            desc = 'A cookie without the HttpOnly flag was sent when ' \
+            keys = ''
+            for key in cookie_obj.keys(): 
+                keys += key + ' '
+            keys = keys.strip()
+            desc = 'Cookie(s) (%s) without the HttpOnly flag was sent when ' \
```

Comment:
I believe that here we have a deep problem that was present in the plugin before you started to work on it, but we just found out about it when trying to implement #5377.

In most cases the application will send only one cookie. In that case we'll find that the message works well:

`Cookie(s) (foobar) without HttpOnly...`

The only problem there is that the message says "Cookie(s)" and the "(s)" is not really needed because there is only one cookie. But then, the real problem occurs... what happens if in one HTTP response we receive two cookies, one with httponly and one without? I believe the code will show:

`Cookie(s) (withouthttponly, withhttponly) without HttpOnly...`

Which is incorrect, since one of the cookies is secure (withhttponly) and we're saying it's not.

Could you please test this?